backend/main/src/Group/Repositories/GroupRepository.ts

```diff
@@ -0,0 +1,154 @@
+import { IGroupRepository } from "src/Group/Domain/IGroupRepository";
+import { Group, GroupId } from "src/Group/Domain/Entities/Group";
+import { NishikiDynamoDBClient } from "src/Shared/Adapters/DB/NishikiTableClient";
+import { RepositoryError } from "src/Shared/Layers/Repository/RepositoryError";
+import { hasError } from "result-ts-type";
+import { ContainerId } from "../Domain/Entities/Container";
+import { GroupName } from "../Domain/ValueObjects/GroupName";
+import { UserId } from "src/User";
+
+interface GroupWithRelationsPrimitiveData {
+	groupId: string;
+	groupName: string;
+	userIds: string[];
+	containerIds: string[];
+}
+/**
+ * Group repository.
+ * The Group item's definition is described in the following document.
+ * @link https://genesis-tech-tribe.github.io/nishiki-documents/project-document/database#Group
+ */
+export class GroupRepository implements IGroupRepository {
+	private readonly nishikiDbClient: NishikiDynamoDBClient;
+
+	/**
+	 * If the client is not specified, the default client is used.
+	 * @param nishikiDbClient
+	 * @throws the data in the repository is invalid.
+	 */
+	constructor(nishikiDbClient?: NishikiDynamoDBClient) {
+		this.nishikiDbClient = nishikiDbClient
+			? nishikiDbClient
+			: new NishikiDynamoDBClient();
+	}
+
+	async find(id: GroupId): Promise<Group | null>;
+	async find(id: ContainerId): Promise<Group | null>;
+	async find(id: GroupId | ContainerId): Promise<Group | null> {
+		const groupData: GroupWithRelationsPrimitiveData = {
+			groupId: "",
+			groupName: "",
+			userIds: [],
+			containerIds: [],
+		};
```

Comment:
I implement the find method by fulfilling the `groupData` object. But it's kinda hard to read, so let me know if you have a good idea that combines GroupDetailData , UserIds, and ContainerIds in a readable way.